### PR TITLE
[WIP] Remove sql_splitter.go

### LINKED ISSUE #1430: Can we remove pkg/sql/sql_splitter.go now?
It seems that after the introduction of the new general parser, we are no longer relying on `splitExtendedSQL`. Can we remove it?

```
$ git grep splitExtendedSQL
pkg/sql/sql_splitter.go:// splitExtendedSQL splits an extended select statement into
pkg/sql/sql_splitter.go:func splitExtendedSQL(slct string) ([]string, error) {
pkg/sql/sql_splitter_test.go:   s, err := splitExtendedSQL(`select a train b with c;`)
pkg/sql/sql_splitter_test.go:   s, err = splitExtendedSQL(`  select a predict b using c;`)
pkg/sql/sql_splitter_test.go:   s, err = splitExtendedSQL(` select a from b;`)
pkg/sql/sql_splitter_test.go:   s, err = splitExtendedSQL(`train a with b;`)
```

### LINKED ISSUE #1447: Recursive call in parse() might overflow the stack
The function `parse` in `/pkg/sql/parse.go` uses the recursion as a loop. 

https://github.com/sql-machine-learning/sqlflow/blob/69c5fe84acf79724416d5c389834e2809e8d5dd2/pkg/sql/parse.go#L106

The recursion would grow the stack size up to the number of statements in an SQL program.  If the SQL program is long enough, this recursive would overflow the stack and crash the SQLFlow server.

We need to replace it with a real loop.

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fix https://github.com/sql-machine-learning/sqlflow/issues/1430

Fix https://github.com/sql-machine-learning/sqlflow/issues/1447

### CODE REVIEW

File: pkg/parser/extended_syntax_parser.y [hunk 1]
```diff
@@ -436,20 +436,19 @@ func (s StandardSelect) String() string {
...
 	defer func() {
 		if r := recover(); r != nil {
-			var ok bool
-			e, ok = r.(error)
-			if !ok {
-				e = fmt.Errorf("%v", r)
-			}
+                        l := r.(*lexer)
+                        idx = l.pos - len(l.recent)
+			e = fmt.Errorf("%v", r)
```
Comment (tonyyang-svail) [inline pkg/parser/extended_syntax_parser.y:444]: Here are some indentation issues. I am wondering why the `go fmt` in CI doesn't catch it.

Comment (wangkuiyi) [reply pkg/parser/extended_syntax_parser.y:444]: Good catch!  This is not a .go file, but a .y file. I will re-indent manually after I fix the CI failure.

**Comment by wangkuiyi** [comment]
Due to the lack of understanding of the related code, I gradually added too many related changes into this PR and made it too big for me to handle. I am closing this PR and will try to split it into smaller pices.

**Comment by wangkuiyi** [comment]
https://github.com/sql-machine-learning/sqlflow/pull/1477 is part of the goal

**Pull request closed without merging**